cpp/open3d/visualization/rendering/Scene.h

```diff
@@ -41,83 +42,110 @@ class AxisAlignedBoundingBox;
-            const std::string& name) = 0;
-    virtual void AssignMaterial(const GeometryHandle& geometry_id,
-                                const MaterialInstanceHandle& material_id) = 0;
-    virtual MaterialInstanceHandle GetMaterial(
-            const GeometryHandle& geometry_id) const = 0;
-    virtual void SetGeometryShadows(const GeometryHandle& geometry_id,
-                                    bool casts_shadows,
-                                    bool receives_shadows) = 0;
-    virtual std::vector<GeometryHandle> FindGeometryByName(
-            const std::string& name) = 0;
-    virtual void RemoveGeometry(const GeometryHandle& geometry_id) = 0;
-
-    virtual LightHandle AddLight(const LightDescription& descr) = 0;
-    // TODO: If possible, add getters
-    virtual void SetLightIntensity(const LightHandle& id, float intensity) = 0;
-    virtual void SetLightColor(const LightHandle& id,
-                               const Eigen::Vector3f& color) = 0;
-    virtual Eigen::Vector3f GetLightDirection(const LightHandle& id) const = 0;
-    virtual void SetLightDirection(const LightHandle& id,
-                                   const Eigen::Vector3f& dir) = 0;
-    virtual void SetLightPosition(const LightHandle& id,
-                                  const Eigen::Vector3f& pos) = 0;
-    virtual void SetLightFalloff(const LightHandle& id, float falloff) = 0;
-    virtual void RemoveLight(const LightHandle& id) = 0;
-
-    // Passing empty id disables indirect lightning
-    virtual void SetIndirectLight(const IndirectLightHandle& id) = 0;
+    // Camera
+    virtual void AddCamera(const std::string& camera_name,
+                           std::shared_ptr<Camera> cam) = 0;
+    virtual void RemoveCamera(const std::string& camera_name) = 0;
+    virtual void SetActiveCamera(const std::string& camera_name) = 0;
+
+    // Scene geometry
+    virtual bool AddGeometry(const std::string& object_name,
+                             const geometry::Geometry3D& geometry,
+                             const Material& material) = 0;
+    virtual bool AddGeometry(const std::string& object_name,
+                             const Model& model) = 0;
+    virtual void RemoveGeometry(const std::string& object_name) = 0;
+    virtual void ShowGeometry(const std::string& object_name, bool show) = 0;
+    virtual void OverrideMaterial(const std::string& object_name,
+                                  const Material& material) = 0;
+    virtual void QueryGeometry(std::vector<std::string>& geometry) = 0;
+    virtual void SetGeometryTransform(const std::string& object_name,
+                                      const Transform& transform) = 0;
+    virtual Transform GetGeometryTransform(const std::string& object_name) = 0;
+    virtual geometry::AxisAlignedBoundingBox GetGeometryBoundingBox(
+            const std::string& object_name) = 0;
+    virtual void OverrideMaterialAll(const Material& material,
```

Comment:
Will the material be shared or will there be copies for each geometry?
As I understand a material defines how we visualize the geometry, right?
Assuming we have a sequence of objects (obj1, ..., obj1000) can we change the material for all of these quickly?